sympy/physics/mechanics/kane.py

```diff
@@ -267,9 +267,8 @@ def _initialize_kindiffeq_matrices(self, kdeqs):
 
     def _form_fr(self, fl):
         """Form the generalized active force."""
-
-        if not iterable(fl):
-            raise TypeError('Force pairs must be supplied in an iterable.')
+        if fl != None and (len(fl) == 0 or not iterable(fl)):
+            raise TypeError('Force pairs must be supplied in an non-empty iterable or None.')
```

Comment:
You can test for this condition with something like:

```python
from sympy.utilities.pytest import raises
raises(lambda: _form_fr('bad input')
```

Or whatever you are trying to check for. I think this would be a ValueError more than a type error, no? I guess it is both.